frontend/site-registry/src/app/features/details/SiteDetails.tsx

```diff
@@ -557,16 +557,15 @@ const SiteDetails = () => {
           </div>
         )}
         <div className="section-details-header row">
-          <div className="d-flex">
-            <BannerDetails />
-            {UserType.External === userType &&
-              snapshot.status === RequestStatus.success &&
-              snapshot.snapshot.data !== null && (
-                <div className="py-2 snapshot">
-                  <span>{`Snapshot Taken: ${new Date(snapshot.snapshot.data[0].created)}`}</span>
-                </div>
-              )}
-          </div>
+          {UserType.External === userType &&
+            snapshot.status === RequestStatus.success &&
+            snapshot.snapshot.data !== null && (
+              <div>
+                <BannerDetails
+                  snapshotDate={`Snapshot Taken: ${new Date(snapshot.snapshot.data[0].created).toString().replace(/\s\([^)]+\)$/, '')}`}
```

Comment:
Please update this code. Only pass the required data from slice. Don't pass the complete snapshot slice. This code can cause error as there is no null check before referencing created.  Also we are already using data-fns package to format the date. Please use formatDateRange / formatDate in utility.ts

Reply:
I am only passing the created date as a prop to banner component, is that an issue? also it has already been checked that the snapshot data shouldnt be null prior to that...May be I am mising something so we can discuss that to have a clear understanding. 
Also the format of date does not match with the output of formatDate function present in utility.ts. I can add this in the utility class and use however cannot use the one already present.

Reply:
1. Use selectors to extract only the necessary parts of the state.  Here I could see snapshot slice is accessed 
   and then you are accessing snapshot.snapshot.data[0]. Instead of this approach you should only access snapshots array in Snapshots slice by using selectors.

2. I see the null check above. But there is no check to see array is empty or not before accessing the first element in array. Add check data.length > 0 before accessing first element.


3. Yes, If the current date helper functions are not matching the required formart. Please add a new method utility.ts so we can use later